woocommerce/woocommerce
# Update eslint parser to @babel/eslint-parser

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #43871 .

Updated the eslint configuration to use `@babel/eslint-parser` parser. This fixed the parsing error:
```
  0:0  error  Parsing error: require() of ES Module 
/woocommerce/woocommerce/node_modules/.pnpm/eslint-scope@7.2.2/node_modules/eslint-scope/lib/definition.js from /woocommerce/woocommerce/node_modules/.pnpm/babel-eslint@10.1.0_eslint@8.55.0/node_modules/babel-eslint/lib/require-from-eslint.js not supported.
Instead change the require of definition.js in /woocommerce/woocommerce/node_modules/.pnpm/babel-eslint@10.1.0_eslint@8.55.0/node_modules/babel-eslint/lib/require-from-eslint.js to a dynamic import() which is available in all CommonJS modules
```

I also fixed many eslint errors and auto-formatted the e2e tests js files. I didn't fix all of them though, there are a lot of uses of waiting for network idle state that I was reluctant to remove without further analysis as it could break the tests unless they are replaced with some web assertions to check for readiness.



### How to test the changes in this Pull Request:

1. Check js linting doesn't throw a parsing error. Example: `pnpm eslint ./plugins/woocommerce/tests/e2e-pw/tests/*.js`
2. Make sure e2e tests are passing.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
